src/decisionengine/framework/engine/DecisionEngine.py

```diff
@@ -98,7 +99,7 @@ def _dispatch(self, method, params):
         return func(*params)
 
     def block_until(self, state, timeout=None):
-        with self.workers.unguarded_access() as workers:
+        with self.channel_workers.unguarded_access() as workers:
```

Comment:
It looks like `block_until` doesn't have any tests... I'm not sure that is in scope for this patch set, but probably should get resolved.

Reply:
Well...should I remove `block_until`?  We don't use it anywhere.

Reply:
I'd say we pull it out.  Since it isn't in use, having it there untested makes it a target for surprises later on.

Reply:
Will do